explainer-content-size.md

```diff
@@ -0,0 +1,34 @@
+# The `content-size` CSS property
+
+This is an explainer for why the `content-size` property is needed. A draft of the spec for this property is [here](http://tabatkins.github.io/specs/css-content-size/). The corresponding spec issue is [here](https://github.com/w3c/csswg-drafts/issues/4229).
+
+`content-size` represents the intrinsic layout sizing of a DOM subtree. Intrinsic sizing is an input to layout of ancestors, and is an important input to the layout constraint algorithms of the web. As one example, if an element has [`width: max-content`](https://drafts.csswg.org/css-sizing-3/#valdef-width-max-content) CSS on it, then it is sized to the [max-content inline size](https://drafts.csswg.org/css-sizing-3/#max-content-inline-size), which is the maximum intrinsic width of elements in the subtree. 
+Currently, such intrinsic sizes are determined implicitly from other factors. For example, a div with `width: 200px` has a `200px` intrinsic width that is communicated upwards in layout algorithms in order to provide sizing constraints to ancestors. Another example is that an image with a `width=200px` attribute will have a `200px` intrinsic width; if it instead has a `src` attribute pointing at an image resource that has a `200px` image resolution width, the image will also have a `200px` intrinsic width once the image is loaded.
+
+This proposal exposes these intrinsic sizes explicitly. (In that sense, one motivation for the `content-size` property could be [explaining](https://extensiblewebmanifesto.org) these algorithms. One practical application could be its use in testing their behavior.
```

Comment:
It would be nice to add a blurb about size containment. Something along the lines of trying not to break the web compat, in the sense that if we do use children for layout (ie no size containment) then we want to rely on that. However, if we can guarantee that children layout doesn't matter (ie size containment) then we can safely use content-size in its stead

Reply:
Added.